components/global/AuthenticatedRoute.tsx

```diff
@@ -1,17 +1,48 @@
-import React from "react";
-import { isAuthenticated } from "../../api/auth";
+import React, {useEffect, useState} from "react";
+import { isAuthenticated, getToken } from "../../api/auth";
 import Router from "next/router";
+import { updateUserStore } from "../../states/user/actions";
+import { useDispatch } from "react-redux";
+import { getUser } from "../../api/user";
 
-interface HeaderProps {
+interface AuthenticatedRouteProps {
   className?: string;
 }
 
-const AuthenticatedRoute: React.FC<HeaderProps> = ({ className }) => {
-  if (!isAuthenticated()) {
-    Router.push("/login");
-  }
+const AuthenticatedRoute: React.FC<AuthenticatedRouteProps> = ({ className }) => {
+  
+  const dispatch = useDispatch();
 
-  return null;
+  const setUser = userData => {
+    dispatch(updateUserStore(userData));
+  };
+
+  const fetchUser = async () => {
+    const userData = await getUser();
+    if(!userData){
+      Router.push('/login');
+    }
+    setHasFetchedUser(false);
+    setUser(userData);
+  };
+
+  const [hasFetchedUser, setHasFetchedUser] = useState(true);
```

Comment:
Move this up, its defined after its used. Not an error but still

Reply:
its not used before defined, the functions are defined
the useEffect is after this

Reply:
In code i mean. `setHasFetchedUser` is used on line 25